### PR TITLE
feat: add VERS rpm versioning scheme support

### LINKED ISSUE #57: feat: add VERS rpm versioning scheme support
## Problem Statement

The VERS specification supports RPM versioning scheme, but go-univers currently only implements RPM ecosystem support without VERS integration. Users cannot use VERS range notation with RPM versions like `vers:rpm/>=1:2.4.1-1.el8|<1:3.0.0`.

## Research

From the [VERS specification](https://github.com/package-url/purl-spec/blob/main/VERSION-RANGE-SPEC.rst), RPM uses a "simplified rpmvercmp version comparison routine" and is similar to Debian in using epochs. The [univers Python implementation](https://github.com/aboutcode-org/univers) includes RPM support in its VERS implementation.

RPM versions follow the format: `[epoch:]version[-release]` where epoch is used for version ordering disambiguation, similar to Debian.

## Current State

- ✅ RPM ecosystem exists at `pkg/ecosystem/rpm/`
- ✅ RPM version parsing and comparison implemented with epoch support
- ❌ No VERS integration for RPM versioning scheme
- ❌ Cannot parse/evaluate ranges like `vers:rpm/>=1:2.4.1-1.el8|<1:3.0.0`

## Implementation Strategy

Following the established pattern used for npm, pypi, go, and maven VERS schemes:

1. **Create `pkg/spec/vers/rpm.go`** with:
   - `rpmContains(constraints []string, version string) (bool, error)` function
   - `intervalToRpmRanges(interval interval) []string` function

2. **Update `pkg/spec/vers/vers.go`**:
   - Add `"rpm": rpmContains` to `schemeToContains` map
   - Add `case "rpm": rangeStrs = intervalToRpmRanges(interval)` to `toRanges` function
   - Update supported ecosystems documentation

3. **Add comprehensive tests** in `pkg/spec/vers/rpm_test.go`

## Success Criteria

- [ ] Support VERS RPM ranges: `vers:rpm/>=1:2.4.1-1.el8`
- [ ] Handle epoch versions correctly: `vers:rpm/>=2:1.0.0|<2:2.0.0`
- [ ] Handle release versions: `vers:rpm/>=2.4.1-1.el8|<2.4.2-1.el9`
- [ ] Support all VERS operators: `>=`, `<=`, `>`, `<`, `=`, `!=`
- [ ] Pass comprehensive test suite covering edge cases
- [ ] CLI integration: `univers vers contains "vers:rpm/>=1:2.4.1" "1:2.5.0-1"`
- [ ] Consistent behavior with existing RPM ecosystem implementation

## Resources

- [VERS Specification](https://github.com/package-url/purl-spec/blob/main/VERSION-RANGE-SPEC.rst)
- [Univers Python Implementation](https://github.com/aboutcode-org/univers)
- [RPM Version Comparison](https://rpm.org/user_doc/version_sorting.html)
- Existing implementations: [npm VERS](https://github.com/alowayed/go-univers/issues/35), [pypi VERS](https://github.com/alowayed/go-univers/issues/36), [go VERS](https://github.com/alowayed/go-univers/issues/37)
- Pattern reference: `pkg/spec/vers/npm.go`

### PR DESCRIPTION
**Pull request by alowayed**
## Summary

This PR implements VERS (Version Range Specification) support for the RPM ecosystem, addressing issue #57.

- ✅ Adds `rpmContains()` and `intervalToRpmRanges()` functions in `pkg/spec/vers/rpm.go`
- ✅ Comprehensive test suite with 30 test cases covering all scenarios
- ✅ Supports epoch versions: `vers:rpm/>=1:2.4.1-1.el8|<1:3.0.0`
- ✅ Supports release versions: `vers:rpm/>=2.4.1-1.el8|<2.4.2-1.el9`
- ✅ All VERS operators: `>=`, `<=`, `>`, `<`, `=`, `!=`
- ✅ CLI integration: `univers vers contains "vers:rpm/>=1:2.4.1" "1:2.5.0-1"`
- ✅ Uses comma-separated constraints like gem and cargo for consistency

### Epoch versions
```bash
univers vers contains "vers:rpm/>=1:2.4.1-1.el8" "1:2.5.0-1.el8" # → true
univers vers contains "vers:rpm/>=2:1.0.0|<=2:2.0.0" "1:5.0.0"     # → false
```

### Complex ranges
```bash
univers vers contains "vers:rpm/>=1:2.4.1|<1:3.0.0|!=1:2.5.0" "1:2.6.0" # → true
```

## Test Plan

- [x] All 30 test cases pass including epoch, release, and error scenarios
- [x] CLI integration verified with epoch and release versions
- [x] Full test suite passes (no regressions)
- [x] Code formatting and linting passes
- [x] Follows established VERS implementation patterns

## References

- Fixes #57
- Follows patterns established in npm (#35), pypi (#36), go (#37) VERS implementations
- [VERS Specification](https://github.com/package-url/purl-spec/blob/main/VERSION-RANGE-SPEC.rst)
- [RPM Version Comparison](https://rpm.org/user_doc/version_sorting.html)

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: pkg/spec/vers/semver.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+	}
+
+	if len(parts) > 0 {
+		// SemVer supports both space and comma-separated, using space-separated like npm
+		return []string{strings.Join(parts, " ")}
+	}
+
+	// Empty interval
+	return []string{}
+}
```
[the remark this answers is not in the document]
Comment (alowayed) [reply pkg/spec/vers/semver.go]: No
